packages/react-storage/src/components/StorageBrowser/views/utils/filePreview/fileType.ts

```diff
@@ -0,0 +1,107 @@
+import type { FileData } from '../../../actions';
+import type { AllFileTypes } from '../../../createStorageBrowser/types';
+import { EXTENSION_MAPPINGS, GENERIC_CONTENT_TYPES } from './const';
+
+export function isGenericContentType(contentType?: string): boolean {
+  if (!contentType || typeof contentType !== 'string') {
+    return true;
+  }
+
+  const normalizedContentType = contentType.toLowerCase().trim();
+
+  return GENERIC_CONTENT_TYPES.has(normalizedContentType);
+}
+
+export function getFileTypeFromContentType(
+  contentType?: string
+): AllFileTypes<any> | null {
+  if (!contentType || typeof contentType !== 'string') {
+    return null;
+  }
+
+  const normalizedContentType = contentType.toLowerCase().trim();
+
+  if (normalizedContentType.startsWith('image/')) {
+    return 'image';
+  }
+
+  if (normalizedContentType.startsWith('video/')) {
+    return 'video';
+  }
+
+  if (normalizedContentType.startsWith('text/')) {
+    return 'text';
+  }
+
+  if (
+    normalizedContentType === 'application/json' ||
+    normalizedContentType === 'application/xml' ||
+    normalizedContentType === 'application/csv'
```

Comment:
this seems too restrictive:
check a list of supported media-types [here](https://www.iana.org/assignments/media-types/media-types.xhtml) as well [this recent PR](https://github.com/aws-amplify/amplify-js/pull/14488)